src/linter.js

```diff
@@ -629,4 +633,23 @@ export default class Linter {
       });
   }
 
+  _markBadwordUsage(filename, fileData) {
+    if (fileData) {
+      const words = fileData.split(' ');
+
+      for (let word of words) {
+        // Filter out none a-Z characters to simplify matching with our
+        // badword list.
+        word = word.toLowerCase().replace(/[^a-zA-Z]/, '');
```

Comment:
As far as I can see, this only filters the first non `a-Z` character. Don't we want to replace all occurrences? (Note that this might have a bigger impact on performance).